disnake/utils.py

```diff
@@ -1121,10 +1127,9 @@ def as_chunks(iterator: _Iter[T], max_size: int) -> _Iter[List[T]]:
 
 def flatten_literal_params(parameters: Iterable[Any]) -> Tuple[Any, ...]:
     params = []
-    literal_cls = type(Literal[0])
     for p in parameters:
-        if isinstance(p, literal_cls):
-            params.extend(p.__args__)
+        if get_origin(p) is Literal:
+            params.extend(_unique(flatten_literal_params(p.__args__)))
```

Comment:
Both of these are fixes for 3.8.

The first one is due to the fact that `type(Literal[0])` on 3.8 is `_GenericAlias`, which also matches other generics:
```py
Python 3.8.10 (default, Mar 15 2022, 12:22:08) [GCC 9.4.0] on linux
>>> from typing import *
>>> type(Literal[0])
<class 'typing._GenericAlias'>
```
on 3.8 this means that `flatten_literal_params([Union[str, int]])` results in `[str, int]` instead of `[Union[str, int]]`, even though it's only meant to flatten `Literal` types

The second one is because versions before 3.9.1 didn't recursively flatten or deduplicate literal parameters, which means we have to do it ourselves:
```py
Python 3.8.10 (default, Mar 15 2022, 12:22:08) [GCC 9.4.0] on linux
>>> from typing import Literal
>>> Literal[1, 2, 1, Literal[2, 3, 4], 5]
typing.Literal[1, 2, 1, typing.Literal[2, 3, 4], 5]
```
```py
Python 3.10.4 (main, Apr  1 2022, 15:56:31) [GCC 9.3.0] on linux
>>> from typing import Literal
>>> Literal[1, 2, 1, Literal[2, 3, 4], 5]
typing.Literal[1, 2, 3, 4, 5]
```